nemo_curator/datasets/doc_dataset.py

```diff
@@ -38,6 +38,9 @@ def __len__(self) -> int:
     def persist(self) -> "DocumentDataset":
         return DocumentDataset(self.df.persist())
 
+    def to_backend(self, backend: Optional[str] = None) -> "DocumentDataset":
+        return DocumentDataset(self.df.to_backend(backend))
+
```

Comment:
Does it make sense to use the `ToBackend` module here or add some of the same checks as the __init__ method from there?

cc: @ryantwolf if you have thoughts.